src/farm/library.cairo

```diff
@@ -50,15 +78,274 @@ namespace CarbonableFarmer {
+        let (share, _) = uint256_unsigned_div_rem(dividend_uint256, total_contract);
+        return (share=share,);
+    }
+
+    func registred_owner_of{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        token_id: Uint256
+    ) -> (address: felt) {
+        // [Check] Uint256 compliance
+        with_attr error_message("CarbonableFarmer: token_id is not a valid Uint256") {
+            uint256_check(token_id);
+        }
+        let (address) = registration_.read(token_id);
+        return (address=address,);
+    }
+
+    //
+    // Externals
+    //
+
+    func start_period{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        unlocked_duration: felt, period_duration: felt
+    ) -> (success: felt) {
+        // [Check] Duration inputs validity
+        with_attr error_message("CarbonableFarmer: Invalid period duration") {
+            assert_nn(period_duration);
+        }
+        with_attr error_message("CarbonableFarmer: Invalid locked duration") {
+            assert_le(unlocked_duration, period_duration);
+        }
+
+        // [Effect] Store period information
+        let (current_time) = get_block_timestamp();
+        start_.write(current_time);
+        period_duration_.write(period_duration);
+        unlocked_duration_.write(unlocked_duration);
+
+        return (success=TRUE,);
+    }
+
+    func stop_period{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        success: felt
+    ) {
+        // [Check] Current period
+        let (current_time) = get_block_timestamp();
+        let (start) = start_.read();
+        let (period_duration) = period_duration_.read();
+        let end = start + period_duration;
+        with_attr error_message("CarbonableFarmer: No current period") {
+            assert_le(current_time, end);
+        }
+
+        // [Effect] Reset period information
+        start_.write(0);
+        period_duration_.write(0);
+        unlocked_duration_.write(0);
+
+        return (success=TRUE,);
+    }
+
+    func deposite{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        token_id: Uint256
+    ) -> (success: felt) {
+        alloc_locals;
+
+        // [Security] Start reetrancy guard
+        ReentrancyGuard._start();
+
+        // [Check] Locked status
+        let (status) = is_locked();
+        with_attr error_message("CarbonableFarmer: deposites are currently locked") {
+            assert status = FALSE;
+        }
+
+        // [Check] Uint256 compliance
+        with_attr error_message("CarbonableFarmer: token_id is not a valid Uint256") {
+            uint256_check(token_id);
+        }
+
+        // [Interaction] Transfer token_id from caller to contract
+        let (carbonable_project_address) = carbonable_project_address_.read();
+        let (caller) = get_caller_address();
+        let (contract_address) = get_contract_address();
+        IERC721.transferFrom(
```

Comment:
I am not sure, but I think the "good practice" does advise to ensure the "caller" is the owner of the `token_id` (because the function is a getter/`view`) before execute the `transferFrom` (as this fucntion is an `external`) ?

Reply:
This verification is done by `transferFrom` function, and also manage the `approved` status.

Reply:
👍